enhancements/microshift/disabling-storage-components.md

```diff
@@ -0,0 +1,244 @@
+---
+title: disabling-storage-components
+authors:
+  - "@copejon"
+reviewers:
+  - "@pacevedom: MicroShift team-lead"
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+  - "@jakobmoellerdev Edge Enablement LVMS Engineer"
+  - "@suleymanakbas91,  Edge Enablement LVMS Engineer"
+approvers:
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+api-approvers:
+  - "None"
+creation-date: 2024-07-11
+last-updated: 2024-07-11
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-856
+see-also:
+  - "enhancements/microshift/microshift-default-csi-plugin.md"
+---
+
+# Disabling Storage Components
+
+## Summary
+
+MicroShift is a small form-factor, single-node OpenShift targeting IoT and Edge Computing use cases characterized by
+tight resource constraints, unpredictable network connectivity, and single-tenant workloads. See
+[kubernetes-for-devices-edge.md](./kubernetes-for-device-edge.md) for more detail.
+
+Out of the box, MicroShift includes the LVMS CSI driver and CSI snapshotter. The LVM Operator is not included with the
+platform, in adherence with the project's [guiding principles](./kubernetes-for-device-edge.md#goals). See the
+[MicroShift Default CSI Plugin](microshift-default-csi-plugin.md) proposal for a more in-depth explanation of the
+storage provider and reasons for its integration into MicroShift. Configuration of the driver is exposed via a config
+file at /etc/microshift/lvmd.yaml. The manifests required to deploy and configure the driver and snapshotter are baked
+into the MicroShift binary during compilation and are deployed during runtime.
+
+LVMD is the node-side component of LVMS and is configured via a config file. If LVMS is started with no LVMD config, the
+process will crash, causing a crashLoopBackoff and requiring user intervention. If the file `/etc/microshift/lvmd.yaml`
+does not exist, MicroShift will attempt to generate a config in memory and provide it to LVMD via config map. If it
+cannot create the config, it will skip LVMS and CSI components instantiation.
+
+Thus, it is already _technically_ possible to disable the storage components, but only via this esoteric functionality.
+Users should be provided a clean UX for this feature and not be forced to learn the inner-workings of MicroShift.
+
+## Motivation
+
+There are variety of reasons a user may not want to deploy LVMS. Firstly, MicroShift is designed to run on hosts with as
+little as 2Gb of memory. Users operating on such small form factors are going to be resource conscious and seek to limit
+unnecessary consumption as much as possible. At idle, the LVMS and CSI components consume roughly 250Mb of memory, about
```

Comment:
This is no longer true on recent LVMS installations, we are looking at 50Mi of Memory. Still might be enough of a reason but since thats 1/5 of the value stated here I think its worth correcting.

Reply:
Does the 50Mi include the CSI controllers? On my idling 4.17 cluster, I have the following resource usage:
```
CONTAINER           NAME                    CPU %               MEM                 DISK                INODES

5804d579b06d7       csi-registrar           0.00                18.68MB             0B                  0
63e0e9298009a       topolvm-controller      0.13                31.13MB             0B                  0
7638fab30a248       csi-resizer             0.00                25.91MB             0B                  0
7ab72a81bdfa4       topolvm-node            0.02                17.5MB              0B                  0
b5270a20b9376       csi-provisioner         0.05                27.82MB             0B                  0
d20b0ed020720       csi-snapshotter         0.00                20.97MB             0B                  0
```

Reply:
This is TopoLVM, not LVMS. A TopoLVM installation is significantly less optimized than LVMS

Reply:
@jakobmoellerdev Testing the changes in openshift/microshift#3535 backs up the 50Mi number. I've amended the proposal to reflect this.

Reply:
Here you're saying that "At idle, the LVMS and CSI components consume roughly 250Mb of memory" but after your updated you've added "With LVMS and the CSI components deployed, the cluster overhead is increased by roughly 50Mb and 0.1% CPU." - I think you probably want to update it here too?

Reply:
Done